chainstate/src/detail/tests/normal_tokens.rs

```diff
@@ -0,0 +1,843 @@
+            Err(BlockError::CheckBlockFailed(
+                CheckBlockError::CheckTransactionFailed(CheckBlockTransactionsError::TokensError(
+                    TokensError::IssueErrorIncorrectMetadataURI(_, _)
+                ))
+            ))
+        ));
+    });
+}
+
+#[test]
+fn token_transfer_test() {
+    utils::concurrency::model(|| {
+        let mut test_framework = TestFramework::default();
+        // Issue a new token
+        let values = vec![OutputValue::Token(TokenData::TokenIssuanceV1 {
+            token_ticker: b"SOME".to_vec(),
+            amount_to_issue: Amount::from_atoms(52292852472),
+            number_of_decimals: 1,
+            metadata_uri: b"https://some_site.meta".to_vec(),
+        })];
+        let block_index =
+            process_token(&mut test_framework, ParentBlock::BestBlock, values.clone())
+                .unwrap()
+                .unwrap();
+        let block = test_framework.block(*block_index.block_id());
+        assert_eq!(block.transactions()[0].outputs()[0].value(), &values[0]);
+
+        // Transfer it
+        let token_id = token_id(&block.transactions()[0]).unwrap();
+        let values = vec![OutputValue::Token(TokenData::TokenTransferV1 {
+            token_id,
+            amount: Amount::from_atoms(123456789),
+        })];
+        let _ = process_token(&mut test_framework, ParentBlock::BestBlock, values)
+            .unwrap()
+            .unwrap();
+
+        // Try to transfer exceed amount
+        let values = vec![OutputValue::Token(TokenData::TokenTransferV1 {
+            token_id,
+            amount: Amount::from_atoms(987654321),
+        })];
+        assert!(matches!(
+            process_token(&mut test_framework, ParentBlock::BestBlock, values),
+            Err(BlockError::StateUpdateFailed(
+                ConnectTransactionError::AttemptToPrintMoney(_, _)
+            ))
+        ));
+
+        // Try to transfer token with wrong id
+        let values = vec![OutputValue::Token(TokenData::TokenTransferV1 {
+            token_id: TokenId::random(),
+            amount: Amount::from_atoms(123456789),
+        })];
+        assert!(matches!(
+            process_token(&mut test_framework, ParentBlock::BestBlock, values),
+            Err(BlockError::StateUpdateFailed(
+                ConnectTransactionError::MissingOutputOrSpent
+            ))
+        ));
+
+        // Try to transfer zero amount
+        let values = vec![OutputValue::Token(TokenData::TokenTransferV1 {
+            token_id,
+            amount: Amount::from_atoms(0),
+        })];
+        assert!(matches!(
+            process_token(&mut test_framework, ParentBlock::BestBlock, values),
+            Err(BlockError::CheckBlockFailed(
+                CheckBlockError::CheckTransactionFailed(CheckBlockTransactionsError::TokensError(
+                    TokensError::TransferZeroTokens(_, _)
+                ))
+            ))
+        ));
+    })
```

Comment:
Same thing here. For all the tests in this file, always end with a working reference case.

Reply:
So, I have here a valid case too at the start of the test, but seems it's a real problem. I will highlight through comments all valid cases in all tests. And I'm going to move them at the end of the tests where it is possible.